query-coordinator/src/main/scala/com/socrata/querycoordinator/rollups/ExpressionRewriter.scala

```diff
@@ -25,7 +25,7 @@ class ExpressionRewriter(val rollupColumnId: (Int) => String,
     * Note that every case here needs to ensure to map every expression recursively
     * to ensure it is either a literal or mapped to the rollup.
     */
-  def apply(e: Expr, r: Analysis, rollupColIdx: Map[Expr, Int]): Option[Expr] = {
+  def apply(e: Expr, r: Analysis, rollupColIdx: Map[Expr, Int],allClausesMatch:Boolean = false): Option[Expr] = {
```

Comment:
I'm not sure if this will work - apply() is called recursively, and so we would need to pass this parameter down the recursion tree, otherwise the window function will not be matched if it's not on the first level of the tree.

We are actually creating a special ExpressionRewriter case - would it be better to make another expression rewriter class derived from the ExpressionRewriter, which will do the window function matching in apply(), and the rest of the cases will be done by the parent? You can then decide which one to call in the rewriteSelection() function.

Reply:
Yup you are right, that's much better